tests/test_port_dpb_vlan.py

```diff
@@ -0,0 +1,235 @@
+import time
+import pytest
+from port_dpb import Port
+from port_dpb import DPB
+
+@pytest.mark.usefixtures('dpb_setup_fixture')
+@pytest.mark.usefixtures('dvs_vlan_manager')
+class TestPortDPBVlan(object):
+    def check_syslog(self, dvs, marker, log, expected_cnt):
+        (exitcode, num) = dvs.runcmd(['sh', '-c', "awk \'/%s/,ENDFILE {print;}\' /var/log/syslog | grep \"%s\" | wc -l" % (marker, log)])
+        assert num.strip() >= str(expected_cnt)
+
+    '''
+    @pytest.mark.skip()
+    '''
+    def test_dependency(self, dvs):
+
+        vlan = "100"
+        p = Port(dvs, "Ethernet0")
+        p.sync_from_config_db()
+        self.dvs_vlan.create_vlan(vlan)
+        #print "Created VLAN100"
+        self.dvs_vlan.create_vlan_member(vlan, p.get_name())
+        #print "Added Ethernet0 to VLAN100"
+        marker = dvs.add_log_marker()
+        p.delete_from_config_db()
+        #Verify that we are looping on dependency
+        time.sleep(2)
+        self.check_syslog(dvs, marker, "Cannot remove port as bridge port OID is present", 1)
+        assert(p.exists_in_asic_db() == True)
+
+        self.dvs_vlan.remove_vlan_member(vlan, p.get_name())
+        time.sleep(1)
```

Comment:
If I'm understanding the flow correctly then I think this sleep can be replaced by waiting for the vlan member to be deleted.

Reply:
This sleep was NOT there to wait for VLAN member to be deleted. Instead, when we remove port from VLAN member, port dependency gets removed and the ports gets deleted. Then we can verify that port is deleted from ASIC DB. This was correct. But anyways, I have introduced "not_exists_in_asic_db()" method in port_dpb.py class which uses the wai_for_deleted_entry() function to verify that port is deleted.

Since we are here, I have a questions for you. I saw that you have introduced field, value match method (wait_for_field_match()). Can we introduce one for just field. Why I am asking is:

Suppose I want to verify that port is added to to PORT_NAME_MAP key in counter DB, I can use it.  Similary wait_for_field_delete_entry().

Reply:
That could work! I'm wondering if it might be cleaner to support "don't cares" or even wild carding rather than adding another method.